### PR TITLE
ensuring client-server sync on cluster colors (SCP-2943)

### PR DESCRIPTION
**Pull request by devonbush**
SCP-2943.  Clicking around, there were sometimes cases where the client and server would get out-of-sync as to the custom colors.  this change ensures the cluster_file_info and custom_colors are always included in the server response, and also adds better nil-safing for edge cases.

TO TEST:
 1. go to a study you have edit permissions
 2. click around a whole bunch customizing colors, cancelling, reset to defaults, set again etc...
 3. check that the plot, legend, and server color state are in sync

### CODE REVIEW

**Review by bistline — APPROVED** [review]
I am able to click around a bunch, and the UI stays in sync for the most part.  Occasionally I end up in a place where after I click "Reset", the re-rendered plot still has the updated colors, but then on refresh everything is back to normal.  I suspect this has something to do with front-end caching, and is not a major concern since the data is not actually being persisted server-side.

File: app/javascript/components/visualization/ScatterPlot.js [hunk 1]
```diff
@@ -119,7 +119,7 @@ function RawScatterPlot({
       })
       store.addNotification(successNotification(`Colors saved successfully`))
       const newScatterData = Object.assign({}, scatterData, {
-        customColors: response.cluster_file_info.custom_colors[annotation.name]
+        customColors: response.cluster_file_info?.custom_colors[scatterData.annotParams.name] ?? {}
```
Comment (ehanna4) [inline app/javascript/components/visualization/ScatterPlot.js:122]: I'm not sure i've seen a double question mark operator before - is this essentially a check that the first part has something and if not assign it to be an empty object?

**Review by eweitz — APPROVED** [review]

File: app/javascript/components/visualization/ScatterPlot.js (hunk 1 above)
Comment (devonbush) [reply app/javascript/components/visualization/ScatterPlot.js:122]: Yup, I just learned about it last week.  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator
